public/login.js

```diff
@@ -0,0 +1,16 @@
+function login(e) {
+  e.target.action = `https://api.core.ac.uk/login_check?continue=${encodeURIComponent(
```

Comment:
If you are extracting JS out of the HTML file, could you please rely on `action` attribute as single source of truth to avoid repeating and potential issue in the future if we decide to change the URL?

Reply:
You can use `URL` for that. You have `URLSearchParams`, so we expect to have URL Web API anyway.